File: src/4node/apply/nodes/summarizeNode.ts
// ─── summarizeNode — 汇总匹配结果 ───────────────────────────────────────────
// 归属：apply 子图
// 输入：state.checkResults（JSON 字符串数组）
// 输出：AI 消息（汇总文本）
// 无外部依赖，纯粹的状态转换

import { AIMessage } from '@langchain/core/messages'
import type { ApplyStateType } from '../../../3state/state.js'

/**
 * 将 LLM 匹配结果汇总为用户可读的文本
 *
 * 流程：
 *   - 解析 checkResults 中的 JSON
 *   - 生成格式化的加分项列表
 *   - 返回 AI 消息
 */
export async function summarizeNode(
  state: ApplyStateType
): Promise<Partial<ApplyStateType>> {
  console.log('--apply:summarize')

  const suggestions = state.checkResults
    .map(r => { try { return JSON.parse(r) } catch { return null } })
    .filter(Boolean)

  if (suggestions.length === 0 || (suggestions[0] as any)?.error) {
    return { messages: [new AIMessage(
      '根据您提供的材料，暂未匹配到符合条件的加分项。请确认材料内容是否完整，或补充更多信息。'
    )] }
  }

  const summary = suggestions.map((s: any) =>
    `• **${s.templateName}** / ${s.ruleName}\n  预计加分：${s.estimatedScore} 分\n  理由：${s.reason}`
  ).join('\n\n')

  return { messages: [new AIMessage(`为您匹配到以下加分项：\n\n${summary}`)] }
}

File: src/nodes/apply/summarize_node.py
"""summarizeNode - 结果汇总节点 - 对应 TS: summarizeNode.ts"""
import json
from typing import Any
from langchain_core.messages import AIMessage
from prompts.templates import NO_MATCH_REPLY


def parse_check_results(check_results: list[str]) -> list[dict]:
    """
    解析 checkResults，提取 suggestions
    """
    suggestions = []
    for r in check_results:
        try:
            parsed = json.loads(r)
            if "error" not in parsed:
                suggestions.append(parsed)
        except json.JSONDecodeError:
            continue
    return suggestions


async def summarize_node(state: dict) -> dict:
    """
    结果汇总节点
    
    归属: apply 子图
    功能: 汇总匹配结果，生成回复
    """
    print("--apply:summarize")
    
    check_results: list[str] = state.get("check_results", [])
    suggestions = parse_check_results(check_results)
    
    if not suggestions:
        # 无匹配
        reply = NO_MATCH_REPLY
    else:
        # 有匹配，生成回复
        reply_lines = ["根据您提供的材料，已为您匹配到以下加分项：", ""]
        for i, s in enumerate(suggestions, 1):
            reply_lines.append(f"{i}. **{s['templateName']}** / {s['ruleName']}")
            reply_lines.append(f"   预计加分: {s['estimatedScore']} 分")
            reply_lines.append(f"   匹配理由: {s['reason']}")
            reply_lines.append("")
        
        reply = "\n".join(reply_lines)
    
    return {
        "messages": [AIMessage(content=reply)]
    }
